### ISSUE TITLE
Conditional AssertThat

### ISSUE THREAD

**Issue by rmmcgr** [jwaliszko/ExpressiveAnnotations#115]
In the following scenario in an ASP.NET MVC app:

```
[RequiredIf("MyStatus == MyStatusEnum.OptionA"]
[AssertThat("DueDate >= Today()"]
public DateTime DueDate {get;set;}
```

The AssertThat is firing when the MyStatus enum is not equal to MyStatusEnum.OptionA.
I would have assumed that the AssertThat should only fire if the RequiredIf returns true?

Or am I getting it wrong?

I also tried:

`[AssertThat("MyStatus == MyStatusEnum.OptionA ? DueDate >= Today() : true")]`

But that didn't work either (the error messages seemed to not like the conditional operator).

In the end I have used the following:

`[AssertThat("DueDate >= Today() || MyStatus != MyStatusEnum.OptionA"]`

**Comment by jwaliszko** [comment]
`DateTime` is a struct, value type. Value of such a type is always non-null, so requirement demand is fulfilled. Use `DateTime?` instead.

Btw: Ternary operator is not yet released. It will be shipped with v2.7.0.

**Comment by jwaliszko** [comment]
I've just pushed a change securing from inappropriately applied requirement attributes: https://github.com/jwaliszko/ExpressiveAnnotations/commit/b4afd1eef0ef9b4a7d486081812ef8d1d3750b12. Such a silent misbehaviours are impossible since now.
